pink/solve_ik.py

```diff
@@ -155,8 +155,9 @@ def solve_ik(
     tasks: Iterable[Task],
     dt: float,
     solver: str,
-    damping: float = 1e-12,
     barriers: Optional[Iterable[Barrier]] = None,
+    damping: float = 1e-12,
+    ignore_limits: bool = False,
```

Comment:
This new parameter looks OK to me. Maybe naming it the other way round (`check_limits: bool = True`) to avoid a double negation? (but that's a nit)

Since we can make API-breaking changes with the upcoming new version, we could also remove the call to `check_limits` from `solve_ik` altogether, and for instance move it to `limits.ConfigurationLimit`. As I remember it, the main motivation for this change was to provide a clearer error message when crossing the limits than the nondescript `assert Delta_q is not None` when the QP is unfeasible.

Reply:
The idea of moving it to `limits.ConfigurationLimit` make sense for me, however my main concern is to give user flexibility to be able to turn off checks for configuration limits. This may help avoiding breaking solution loop once the some joints are outside of their respective limits if this is not the issue for the user

Reply:
I have changed this logic, now:
- The `check_limits` method now includes an optional `safe_break` argument to control whether execution should stop on exception.
 - The solve_ik function now includes the `safe_break` that is feed-forwarded to `check_limits`.